check_diff/src/lib.rs

```diff
@@ -117,33 +137,18 @@ impl RustfmtRunner {
             config_arg.as_str()
         );
 
-        // walks the "." directory and finds all files with .rs extensions
-        for entry in WalkDir::new(".").into_iter().filter_map(|e| e.ok()) {
-            let path = entry.path();
-            if path.is_file() && path.extension().map_or(false, |ext| ext == "rs") {
-                let file = OpenOptions::new()
-                    .write(true)
-                    .append(true)
-                    .open(output_path)?;
-                let Ok(_) = Command::new(&self.binary_path)
-                    .env("LD_LIBRARY_PATH", &self.ld_library_path)
-                    .args([
-                        "--unstable-features",
-                        "--skip-children",
-                        "--check",
-                        "--color=always",
-                        config.as_str(),
-                    ])
-                    .stdout(file)
-                    .output()
-                else {
-                    return Err(CheckDiffError::FailedWritingToFile(
-                        "Failed to write to file",
-                    ));
-                };
-            }
-        }
-        Ok(())
+        let output = Command::new(&self.binary_path)
+            .env("LD_LIBRARY_PATH", &self.ld_library_path)
+            .args([
+                "--unstable-features",
+                "--skip-children",
+                "--emit=stdout",
+                config.as_str(),
+                code,
+            ])
+            .output()?;
```

Comment:
I think we'll need to tweak this slightly. We want to pass the `code` to rustfmt via `stdin`, and in order to do that we probably need to do [something similar to `rust-analyzer`](https://github.com/rust-lang/rust-analyzer/blob/c0bbbb3e5d7d1d1d60308c8270bfd5b250032bb4/crates/rust-analyzer/src/handlers/request.rs#L2313-L2324).

Also, I think we could simplify creating the `config` arg above. We should probably abstract that into a standalone function.